src/v/storage/api.h

```diff
@@ -47,7 +48,7 @@ class api {
 
 private:
     std::function<kvstore_config()> _kv_conf_cb;
-    log_config _log_conf;
+    std::function<log_config()> _log_conf_cb;
```

Comment:
I don't think that noncopyable_function exists solely to avoid copies. I think std::function also comes with some overheads like heap allocation when used in certain ways. We should probably use noncopyabe_function, but even better would be to fully understand the differences and I haven't found any specific information in the git history of seastar to explain it.